### DISCUSSION TITLE
A client can boot from the LTSP server if they are in different VLANs?

### DISCUSSION THREAD

**Discussion by lucasoliveira7** [ltsp/ltsp#257] [Q&A]
Hello!

	Guys, i am trying to install an LTSP server in my enterpise, that we have about 600 clients. 
	
	Our infrastructure has about 35 VLANs, for example:
	
	10.10.2.0
	10.10.170.0
	10.10.120.0.....
	
	I installed the LTSP inside the VLAN 10.10.170.0 and worked when the client was in the same ip range, but when i tried to boot from another range (like 10.10.120.0), the client couldn`t find the image from the server and initialize the IPXE boot.
	
	So, i tried to install a clone from the original LTSP in the VLAN 10.10.120.0 and worked normally. I had spend many time serching how can i solve this, but without success. Is there a whay to make a lot of VLANs find the LTSP in the network even if the server is in different VLAN of the client?

	Consider this scenario:

	Our dhcp is distributed from a firewall, that has the ip 172.32.0.1
	
	The LTSP is in a VLAN, with the ip range: 10.10.170.0

	And we have a lot of clients with diferents VLANS that need to boot at the LTSP server.

	Ps: Every single VLANs has rules to comunicate between each other and work normally to other stuff.
		
	If someone can help me, i will be very thankfull! If not, i appreciate as well!

**Comment by maccadia** [comment]
We do use LTSP(5) across multiple VLANs. If you can ping the LTSP server from your client VLANs, you probably have some firewall rules to add.

You need to figure out which ports to open depending on your LTSP configuration. As you probably know, LTSP use several protocols : tftp, dhcp, nfs, ssh ; each one need specific firewall settings.

**Comment by lucasoliveira7** [comment]
I am using LTSP(19) and our VLANs has all traffic allowed between them, so i have to configure this ports on my LTSP server? Because in my Firewall is all allowed...

**Comment by maccadia** [comment]
If everything is allowed on your firewall. I don't know where is the issue. Maybe a conflict between dnsmasq and your dhcp server.

You said that the clients can't find the iPXE binary to boot from. So this is probably a DHCP issue. What is the exact error ?

You can look at https://ltsp.org/advanced/isc-dhcp-server/ if you have isc-dhcp.

**Comment by alkisg** [comment]
@lucasoliveira7, you need to paste the client error message so that we see what's wrong.

Are you using the default proxyDHCP setup, or only the main DHCP?

What is the output of this command, on an LTSP server or client? Replace enp2s0 with an existing interface.

```shell
/usr/lib/klibc/bin/ipconfig -n enp2s0
```

**Comment by lucasoliveira7** [comment]
The output of this command: 

root@LTSP:/home# /usr/lib/klibc/bin/ipconfig -n enp2s0
IP-Config: enp2s0 hardware address d0:94:66:b6:85:00 mtu 1500 DHCP RARP
IP-Config: enp2s0 guessed broadcast address 10.10.170.255
IP-Config: enp2s0 complete (dhcp from 10.10.170.1):
 address: 10.10.170.17     broadcast: 10.10.170.255    netmask: 255.255.255.0   
 gateway: 10.10.170.1      dns0     : 10.10.2.100      dns1   : 10.10.2.3       
 rootserver: 10.10.170.17 rootpath: 
 filename  :

**Comment by lucasoliveira7** [comment]
@alkisg 

i'm using the default proxyDHCP config.

**Comment by alkisg** [comment]
I do not have experience with VLANs and you didn't paste the client error message.

I would suggest to boot a client with some iPXE version from https://boot.ipxe.org, for example the .usb or the .iso image. Press Ctrl+B when prompted, then run "autoboot". I'm guessing it will fail.

At that point, type `config` in the iPXE boot prompt, and see if you got a proxyDHCP reply.

At the same time, run `journalctl -fu dnsmasq` on the LTSP server, and see if you get a line like this:

```shell
Mar  6 09:03:32 alkis dnsmasq-dhcp[5996]: PXE(enp2s0) 00:d8:61:98:11:85 proxy
```

If you don't see a line like that, and iPXE reports no proxy offers, then the problem is that the LTSP server isn't notified about the client DHCP request. That broadcasted traffic (it's Ethernet traffic, not IP traffic yet) needs to reach the proxyDHCP server.

**Comment by lucasoliveira7** [comment]
Sorry,

The client had the follow problem: 

PXE-E32: TFTP Open timeout

**Comment by lucasoliveira7** [comment]
@alkisg 

I will try the possibilities that you and @MAccadia sugest

Thank you!

**Comment by lucasoliveira7** [comment]
Thatś what i see when i run: journalctl -fu dnsmasq

mar 05 17:04:00 LTSP dnsmasq-dhcp[12793]: PXE(enp2s0) d0:94:66:db:8e:07 proxy

**Comment by alkisg** [comment]
And does iPXE on the client get that proxy offer?
I.e. something like this image from ipxe.org?

![ipxe config>proxydhcp](http://i.imgur.com/CgiVwab.png "ipxe config>proxydhcp")

**Comment by lucasoliveira7** [comment]
When the client is in a different VLAN i can't see the ipxe menu, the client show me this:
![ltspclient](https://user-images.githubusercontent.com/61873280/76110774-e560b780-5fbd-11ea-9c3a-3d06657e85b7.PNG)

**Comment by alkisg** [comment]
Please read again my previous message about creating a USB stick or CD with iPXE, from boot.ipxe.org.

**Comment by lucasoliveira7** [comment]
I tried and that is what i got, this time the the client find the ltsp server

![clientIPXE](https://user-images.githubusercontent.com/61873280/76112756-9ae13a00-5fc1-11ea-8ef5-1e383a97660f.PNG)

**Comment by alkisg** [comment]
Check if TFTP traffic is allowed between VLANs.

**Comment by lucasoliveira7** [comment]
@alkisg 

It work!!

thank you very much, i was hitting my head becaus of it for at least a week.

Just to clarify, at this time i was testing in a vlan that didn't had all the traffic released in the VLAN that LTSP is in, my mistake.

So, the boot across VLANs will only work if i have a USB with iPXE installed to boot?  And why this isn't required when the client and the server are in the same VLAN?

**Comment by alkisg** [comment]
No, we only used iPXE because it helps us troubleshoot things.

What happens now that you unblocked TFTP, if you use the BIOS PXE boot?

**Comment by lucasoliveira7** [comment]
The bios PXE boot don't find nothing, looks like the client search and search but without success, seems like there is nothing to boot

**Comment by alkisg** [comment]
VLAN somehow blocks the proxy DHCP traffic. Try to unblock the PXE port, 4011.

**Comment by lucasoliveira7** [comment]
The VLANs has all tfraffic released between each other, that is what i can't understand....

**Comment by maccadia** [comment]
I don't know why proxy DHCP is not working across your VLANs. I found [this](https://community.arubanetworks.com/t5/Wired-Intelligent-Edge-Campus/PXE-Boot-TFTP-across-VLANs-with-Aruba-5412Rzl2/td-p/544507) but I don't know what are ip helpers.

Maybe you have some [DHCP snooping](https://en.wikipedia.org/wiki/DHCP_snooping) configured on your switches that prevent DHCP requests to pass across VLANs.

In our setup, we don't use proxy DHCP. We configure our DHCP server to send DHCP options 66 (next-server) and 67 (bootfile name).

**Comment by lucasoliveira7** [comment]
@alkisg 

The VLANs has all traffic between each other, i don't see where else i can allow something more....

**Comment by lucasoliveira7** [comment]
I will try an bios update, maybe can fix the a BIOS PXE problema, who knows...

**Comment by lucasoliveira7** [comment]
@MAccadia 

My setup is the same as yours the server is sending 66 and 67 options too...

About the DHCP Snooping, i don't know nothing about it, i will ask some people here.

I search about ip helpers and i saw this https://www.ciscopress.com/articles/article.asp?p=330807&

**Comment by alkisg** [comment]
@lucasoliveira7, a DHCP request has IP traffic but it also has Ethernet traffic.

I do not know if you can forward Ethernet traffic with VLANs automatically or if you need some special config or if you need DHCP helpers etc.

I do not have a VLAN setup to test with, so I cannot help you more via this issue.

Either google, or wait someone that knows about VLANs, or come to IRC to troubleshoot this via IRC/VNC/VM client.

**Comment by maccadia** [comment]
> My setup is the same as yours the server is sending 66 and 67 options too...

In this case, you should disable proxy dhcp on the LTSP server with `ltsp dnsmasq --proxy-dhcp=0 --real-dhcp=0`.

I see two potential issues : 

1. Do your clients get the correct next-server and bootfile name ? (you can see them with iPXE on USB ; go to configuration > net0 > dhcp) What are they ?
2. Does TFTP works from your clients' VLAN to your server ? e.g from a ubuntu laptop, install tftp (`sudo apt install tftp`) and try to download something : 
```bash
$ tftp [server_ipaddress]
tftp > get [tftp_file_name]
tftp > quit
```
